packages/graphql/src/translate/queryAST/ast/fields/attribute-fields/CypherUnionAttributePartial.ts

```diff
@@ -51,7 +51,17 @@ export class CypherUnionAttributePartial extends QueryASTNode {
     }
 
     public getFilterPredicate(variable: Cypher.Variable): Cypher.Predicate {
-        return (variable as Cypher.Node).hasLabels(...this.target.getLabels());
+        const labels = this.target.getLabels();
+
+        // TODO: Refactor when `.hasLabel` on variables is supported in CypherBuilder
+        const predicates = labels.map((label) => {
+            return new Cypher.RawCypher((env) => {
+                const varName = env.compile(variable);
+                const labelStr = Cypher.utils.escapeLabel(label);
+                return `${varName}:${labelStr}`;
+            });
+        });
```

Comment:
If I understood correctly the reason behind maybe the following works as well!

```suggestion
        const predicates = labels.map((label) => {
            return (variable as Cypher.Node).hasLabel(label);
        });
```

Reply:
It works, but it is unsafe, as we do not guarantee that a variable is a Node at the moment